app.rb

```diff
@@ -0,0 +1,196 @@
+    @io.puts "\n"
+    case input
+    when "1"
+      list_items
+    when "2"
+      create_item
+    when "3"
+      list_orders
+    when "4"
+      create_order
+    when "5"
+      exit
+    else 
+      @io.puts "Invalid! Please enter valid option.".colorize(:red)
+    end
+  end
+
+  def list_items
+    @io.puts "ITEMS IN STOCK:".colorize(:blue)
+    @io.puts "\n"
+    items = @item_repository.all
+    items.each do |record|
+      @io.puts "##{record.id} #{record.name} - Unit price: #{record.price} - Quantity: #{record.quantity}"
+    end
+  end
+
+  def create_item
+    item = Item.new
+    item.name = get_item_name
+    item.price = get_price
+    item.quantity = get_quantity
+    
+    @item_repository.create(item)
+    @io.puts "#{item.name} was added in stock.".colorize(:green)
+  end
+
+  def list_orders
+    @io.puts "CURRENT ORDERS:".colorize(:blue)
+    @io.puts "\n"
+    
+    get_id_list.each do |order_id|
+      order = @order_repository.find_by_order(order_id)
+      @io.puts "##{order.id} #{order.customer_name} - Order date: #{order.order_date} - Items:"
+      order.items.each do |item|
+        @io.puts "- ##{item.id} #{item.name} - Price: #{item.price}"
+      end
+      @io.puts "\n"
+    end
+  end
+
+  def create_order
+    order = Order.new
+    order.customer_name = get_customer_name
+    order.order_date = get_date
+    item_id = get_linked_item
+
+    @order_repository.create(order)
+
+    order_id = @order_repository.all.last.id
+
+    @order_repository.link_to_item(order_id, item_id)
+
+    @io.puts "Order for #{order.customer_name} was created.".colorize(:green)
+
+    item = @item_repository.find(item_id)
+    
+    # Reduces item quantity by 1 after the order is placed.
+    reduce_stock(item)
+  end
+
+  def reduce_stock(item)
+    item.quantity = item.quantity.to_i - 1
+    @item_repository.update(item)
+
+    @io.puts "#{item.name} quantity in stock was reduced by one.".colorize(:green)
+  end
+
+  private
+
+  def get_id_list
+    id_list = []
+    @order_repository.all.each do |record| 
+      id_list << record.id
+    end
+    id_list
+  end
+
+  def get_item_name
+    @io.puts "Enter item name:"
+    item_name = @io.gets.chomp 
+  end
+
+  def get_price
+    @io.puts "Enter item price:"
+    item_price = @io.gets.chomp
+    
+    while item_price.to_f <= 0 # Loops if anything but positive number was entered
+      @io.puts "Please enter valid price.".colorize(:red)
+      item_price = @io.gets.chomp
+    end
+  
+    item_price.to_f
+  end
+
+  def get_quantity
+    @io.puts "Enter item quantity:"
+    item_quantity = @io.gets.chomp
+    
+    while (item_quantity.to_i.to_s != item_quantity) || (item_quantity.to_i <= 0)
+      @io.puts "Please enter valid quantity.".colorize(:red)
+      item_quantity = @io.gets.chomp
+    end
+    
+    item_quantity.to_i
+  end
+
+  def get_customer_name
+    @io.puts "Enter customer name:"
+    customer_name = @io.gets.chomp
```

Comment:
this is super minor, but could you add an auto-capitalization element in here so if someone enters lower case, it's automatically changed?